.github/workflows/on-push.yml

```diff
@@ -0,0 +1,70 @@
+name: Push
+
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: volta-cli/action@v1
+      - name: Cache node_modules
+        id: cache-primes
+        uses: actions/cache@v1
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-${{hashFiles('**/package-lock.json')}}-node_modules
+      - run: npm install
+      - run: npm run lint
+      - run: npm run build
+
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - "5432:5432"
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis
+        ports:
+          - "6379:6379"
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: volta-cli/action@v1
+      - name: Cache node_modules
+        id: cache-primes
+        uses: actions/cache@v1
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-${{hashFiles('**/package-lock.json')}}-node_modules
+      - run: npm install
+      - run: npm run db:test
+      - name: Run tests
+        env:
+          NODE_ENV: test
+          REDIS_HOST: localhost
+          REDIS_PORT: ${{ job.services.redis.ports[6379] }}
```

Comment:
What does this syntax mean? If we use the port 6379 as an index, is there any difference to just hard coding the port into the `REDIS_PORT` env var?

Reply:
It's how [github recommends to do it based on their example](https://github.com/actions/example-services/blob/main/.github/workflows/redis-service.yml).  This syntax is just to get the associated public port based on the internal port.  In case the public port changes (conflicts?), it'll make sure that it won't affect the test.